game-app/game-core/src/main/java/games/strategy/engine/framework/ServerGame.java

```diff
@@ -441,11 +441,29 @@ private void runStep(final boolean stepIsRestoredFromSavedGame) {
               ? launchAction.getAutoSaveFileUtils().getEvenRoundAutoSaveFile()
               : launchAction.getAutoSaveFileUtils().getOddRoundAutoSaveFile());
     }
-    if (autoSaveThisDelegate && !currentStep.getName().endsWith("Move")) {
+    if (shouldAutoSaveAfterEnd(currentDelegate) && !currentStep.getName().endsWith("Move")) {
```

Comment:
Probably out of scope, but didn't we have convenience methods for combat and non-combat moves here?
Could we introduce them for "Move" as well to prevent potential typos?

Reply:
We do, done!